serpent/src/lib.rs

```diff
@@ -77,49 +77,34 @@ fn linear_transform_inverse_bitslice(mut words: Words) -> Words {
     words
 }
 
-fn apply_s(index: usize, nibble: u8) -> u8 {
-    S[index % 8][nibble as usize]
-}
-
-fn apply_s_inverse(index: usize, nibble: u8) -> u8 {
-    S_INVERSE[index % 8][nibble as usize]
-}
-
+#[inline]
 fn apply_s_bitslice(index: usize, [w1, w2, w3, w4]: Words) -> Words {
-    let mut words = [0u32; 4];
-
-    for i in 0..32 {
-        let quad = apply_s(
-            index,
-            get_bit(w1 as usize, i)
-                | (get_bit(w2 as usize, i) << 1)
-                | (get_bit(w3 as usize, i) << 2)
-                | (get_bit(w4 as usize, i) << 3),
-        );
-
-        for l in 0..4 {
-            words[l] |= u32::from(get_bit(quad as usize, l)) << i;
-        }
+    match index % 8 {
```

Comment:
It's probably worth to move `apply_s(_inverse)_bitslice` and `linear_transform(_inverse)_bitslice` to the bitslice module. It would allow us to make the `sbox_*` functions private to the module.

Reply:
Done. I also chopped `_bitslice` off the names since it seems redundant with the module name prefix.